### PR TITLE
Add agent config `PROPERTIES_REPORT_PERIOD_FACTOR `, Avoid the instanceTraffic record properties is null.

### PR DESCRIPTION
**Pull request by zifeihan**
…ecord properties is null.

Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bug fix
- [x] New feature provided
- [ ] Improve performance

- Related issues

___
### Bug fix
- Bug description.

- How to fix?

___
### New feature or improvement
- Describe the details and related test reports.
because instanceTraffic record can be cleaned and the agent properties only send once. so some  instanceTraffic record properties is null, Add agent config `TIMES_SEND_PROPERTIES` for resend agent properties.

### CODE REVIEW

File: apm-sniffer/optional-reporter-plugins/kafka-reporter-plugin/src/main/java/org/apache/skywalking/apm/agent/core/kafka/KafkaServiceManagementServiceClient.java [hunk 1]
```diff
@@ -85,28 +87,32 @@ public void boot() {
...
+            producer.flush();
+        } else {
+            InstancePingPkg ping = InstancePingPkg.newBuilder()
+                                                  .setService(Config.Agent.SERVICE_NAME)
+                                                  .setServiceInstance(Config.Agent.INSTANCE_NAME)
+                                                  .build();
+            if (LOGGER.isDebugEnable()) {
+                LOGGER.debug("Heartbeat reporting, instance: {}", ping.getServiceInstance());
+            }
+            producer.send(new ProducerRecord<>(topic, ping.getServiceInstance(), Bytes.wrap(ping.toByteArray())));
```
Comment (kezhenxu94) [inline apm-sniffer/optional-reporter-plugins/kafka-reporter-plugin/src/main/java/org/apache/skywalking/apm/agent/core/kafka/KafkaServiceManagementServiceClient.java:114]: Why do you move this `ping` into the `else` block? This should be as what it was.

File: docs/en/setup/service-agent/java-agent/README.md [hunk 2]
```diff
@@ -89,6 +89,7 @@ property key | Description | Default |
 `osinfo.ipv4_list_size`| Limit the length of the ipv4 list size. |`10`|
 `collector.grpc_channel_check_interval`|grpc channel status check interval.|`30`|
 `collector.heartbeat_period`|agent heartbeat report period. Unit, second.|`30`|
+`collector.times_send_properties`|How many times do report a properties to the backend, Each time interval is {@link Collector#HEARTBEAT_PERIOD} |`10`|
```
Comment (kezhenxu94) [inline docs/en/setup/service-agent/java-agent/README.md:92]: It's hard to express this config item properly, but this statement is abs. not what you really mean, according to the logic, it should be

```suggestion
`collector.times_send_properties`|The agent sends the instance properties to the backend every `collector.heartbeat_period * collector.times_send_properties` seconds |`10`|
```

Comment (wu-sheng) [reply docs/en/setup/service-agent/java-agent/README.md:92]: I prefer `collector.times_send_properties` to `collector.properties_report_period_factor`

File: apm-sniffer/optional-reporter-plugins/kafka-reporter-plugin/src/main/java/org/apache/skywalking/apm/agent/core/kafka/KafkaServiceManagementServiceClient.java (hunk 1 above)
Comment (zifeihan) [reply apm-sniffer/optional-reporter-plugins/kafka-reporter-plugin/src/main/java/org/apache/skywalking/apm/agent/core/kafka/KafkaServiceManagementServiceClient.java:114]: The KafkaServiceManagementServiceClient need sends the instance properties to the backend every `collector.heartbeat_period * collector.times_send_properties` seconds

Comment (zifeihan) [reply apm-sniffer/optional-reporter-plugins/kafka-reporter-plugin/src/main/java/org/apache/skywalking/apm/agent/core/kafka/KafkaServiceManagementServiceClient.java:114]: @kezhenxu94 KafkaServiceManagementServiceClient also need to send the instance properties.

File: apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/conf/Config.java [hunk 3]
```diff
@@ -142,6 +142,10 @@
          * The period in which the agent report a heartbeat to the backend.
          */
         public static long HEARTBEAT_PERIOD = 30;
+        /**
+         * The agent sends the instance properties to the backend every `collector.heartbeat_period * collector.times_send_properties` seconds
```
Comment (kezhenxu94) [inline apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/conf/Config.java:146]:
```suggestion
         * The agent sends the instance properties to the backend every `collector.heartbeat_period * collector.properties_report_period_factor` seconds
```

File: apm-sniffer/optional-reporter-plugins/kafka-reporter-plugin/src/main/java/org/apache/skywalking/apm/agent/core/kafka/KafkaServiceManagementServiceClient.java (hunk 1 above)
Comment (wu-sheng) [reply apm-sniffer/optional-reporter-plugins/kafka-reporter-plugin/src/main/java/org/apache/skywalking/apm/agent/core/kafka/KafkaServiceManagementServiceClient.java:114]: I think the question is, ping should be sent every time, no `else`. You should not change this part of the original codes.

**Review by kezhenxu94 — COMMENTED** [review]
LGTM basically

**Review by kezhenxu94 — APPROVED** [review]

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**